spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -122,7 +127,39 @@ public ReadWriteConversions getConversions() {
 			throw new DatastoreDataException("Unable to read " + persistentEntity.getName() + " entity", ex);
 		}
 
-		return instance;
+		return (R) instance;
+	}
+
+	private DatastorePersistentEntity getDiscriminationPersistentEntity(DatastorePersistentEntity ostensibleEntity,
+			EntityPropertyValueProvider propertyValueProvider) {
+		if (ostensibleEntity.getDiscriminationFieldName() == null) {
+			return ostensibleEntity;
+		}
+		DatastorePersistentEntity persistentEntity = null;
+
+		Set<Class> members = DatastoreMappingContext.getDiscriminationFamily(ostensibleEntity.getType());
+
+		if (members != null) {
+			for (Class member : members) {
+				DatastorePersistentEntity memberEntity = this.mappingContext.getPersistentEntity(member);
+				if (memberEntity != null && isDiscriminationFieldMatch(memberEntity, propertyValueProvider)) {
+					if (persistentEntity != null) {
+						throw new DatastoreDataException(
```

Comment:
Can we validate that when DiscriminationFamily map is created? That would be more efficient.